File: P/1103_Distribute_Candies_to_People.java
//Brute Force
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int[] ans = new int[num_people];
        
        int turn = 0;
        int candy = 1;
        while(candies > 0) {
            ans[turn] += candies - candy >= 0 ? candy : candies;
            candies -= candy++;
            turn = (turn + 1) % num_people;
        }
        return ans;
    }
}

File: 주영/Any/1103. Distribute Candies to People.js
/**
 * @param {number} candies
 * @param {number} num_people
 * @return {number[]}
 */
var distributeCandies = function (candies, num_people) {
  let arr = new Array(num_people).fill(0);
  let i = 0;

  while (candies - i > 0) {
    arr[i % num_people] += i + 1;
    i++;
    candies -= i;
  }
  //   console.log('i: ', i);
  //   console.log('candies: ', candies);
  //   console.log('arr: ', arr);

  arr[i % num_people] += candies;

  return arr;
};

let candies = 7,
  num_people = 4;
distributeCandies(candies, num_people);
